### ISSUE TITLE
Favorite star inconsistencies

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly#3674]
Reviewing app launcher, select, table, and menu, I found these discrepancies in the hover states (the focus styles match the hover styles):

App launcher:
* Hovering either a favorited or non-favorited star turns the star black.

Select and Table:
* Hovering over a non-favorited star turns the star black
* Hovering over a favorited star, the star remains gold

Menu:
* No hover styles

----

Also all of the favorite buttons when disabled do not change colors except for the select - the select component updates the start with the disabled text color, which looks a little odd.

<img width="892" alt="Screen Shot 2020-11-13 at 3 52 03 PM" src="https://user-images.githubusercontent.com/35148959/99124703-47030800-25c8-11eb-835f-8d6de6c6e84b.png">

@mcarrano @mceledonia can you confirm what these should be? Should we take this opportunity to also introduce a favorite component that is a button, the icon, and the hover, focus, and disabled states built in?

**Comment by mcarrano** [comment]
Yes, we were recently discussing this. @mceledonia can you add the desired behavior here? I agree that we should create a Favorites component to use consistently. I'll look at getting this on the roadmap.

**Comment by mceledonia** [comment]
@mcoker @mcarrano 

For all favorite interaction across all components - 

**Hover**
- Hovering an unfavorited star (gray) should result in a black star
- Hovering a favorited star (gold) should result in a darker gold star `$pf-color-gold-500`
